### PR TITLE
Fix auction id transaction

### PR DESCRIPTION
**Pull request by sunce86**
# Description
Fixes https://github.com/cowprotocol/services/issues/1956

Replaces priority when searching for `auction_id`. Now first check if tx has `auction_id` appended, if not, search from database. This way, tx mined with colocation enabled will not be associated with wrong `auction_id`.

Note that I did not completely erase searching from database because:
1. We still have 20minutes of non-colocated transactions to index the moment we switch to colocation
2. We still want to be able to go back in past and reindex old non-colocated transactions.

### CODE REVIEW

File: crates/autopilot/src/on_settlement_event_updater.rs [hunk 1]
```diff
@@ -113,17 +113,19 @@ impl OnSettlementEventUpdater {
...
+            .await?
+            .map(AuctionId::Colocated);
         if auction_id.is_none() {
             // This settlement either belongs to the other environment (staging, prod) or it
-            // was issued using solver-driver colocation. In that case solvers
-            // are supposed to append the `auction_id` to the calldata.
-            auction_id = Self::recover_auction_id_from_calldata(&mut ex, &transaction)
-                .await?
-                .map(AuctionId::Colocated);
+            // was issued BEFORE solver-driver colocation.
```
Comment (MartinquaXD) [inline crates/autopilot/src/on_settlement_event_updater.rs:121]: The comment is now slightly incorrect. Not finding the auction id on the calldata is not related to prod/staging. The only thing we can tell is that at this point is that the auction was issued pre-colocation.
But we could also adjust the current `solver` code to append the auction id as well. I **believe** that should be fine.

**Review by MartinquaXD — APPROVED** [review]

**Review by fleupold — APPROVED** [review]

**Review by nlordell — APPROVED** [review]

File: crates/autopilot/src/on_settlement_event_updater.rs (hunk 1 above)
Comment (sunce86) [reply crates/autopilot/src/on_settlement_event_updater.rs:121]: I initially did this, but reverted the commit, for two reasons:
1. Fix works without appending the `auction_id` to current `solver`.
2. If we do it, then we lose the distinction between `Centralized` and `Colocated` auction ids in the `on_settlement_event_updater.rs` which makes thing really confusing and hard to debug

Comment (sunce86) [reply crates/autopilot/src/on_settlement_event_updater.rs:121]: Will adjust the comment

Comment (sunce86) [reply crates/autopilot/src/on_settlement_event_updater.rs]: Fixed.

**Pull request merged**